File: 162-find-peak-element.go
func findPeakElement(nums []int) int {
    if len(nums) == 1{
        return 0
    }
    left, right := 0, len(nums) - 1
    for left <= right{
        mid := (left + right) / 2
        v := nums[mid]
        if left == right{
            return left
        } else if v < nums[mid+1] {
            left = mid + 1
        } else {
            right = mid
        }
    }
    return left
}

File: bwu/binary_search/162-find-peak-element.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        left, right = 0, len(nums) - 1
        while left + 1 < right:
            mid = (left + right) / 2
            if nums[mid] > nums[mid + 1] and nums[mid] > nums[mid - 1]:
                return mid
            elif nums[mid] < nums[mid + 1]:
                left = mid
            else:
                right = mid
                
        return left if nums[left] > nums[right] else right
